### DISCUSSION TITLE
Passing command keys to Neovim

### DISCUSSION THREAD

**Discussion by notahat** [ghostty-org/ghostty#3673] [Feature Requests, Ideas]
Hi folks.

Congratulations on the 1.0 release, and thanks for all the hard work that has obviously gone into it. Looks really great so far! The performance, the sensible default config, and the quality font rendering are all impressive.

This post is a feature request for a thing I use in Kitty that (I think) Ghostty can't do right now: passing Mac command keys through to Neovim, but _only_ when Neovim is open.

This is particularly useful for making command-X, command-C, and command-V work with the system clipboard in both the shell and Neovim.

(To be clear: I'm pretty sure I can configure Ghostty to _always_ pass Command keys through by removing the built-in keybinds for them. It's the doing it _only when Neovim is open_ that I think Ghostty can't do currently.)

I realise this might be a niche feature, but it at least aligns with Ghostty's goal of providing a rich set of terminal features, and might be something that wasn't on your radar before. :heart:

---

So how does Kitty handle this?

The Kitty docs have [the recipe](https://sw.kovidgoyal.net/kitty/mapping/#conditional-mappings-depending-on-the-state-of-the-focused-window), but I'll describe my setup here for reference:

First, I get Neovim to tell Kitty when it's open in my Neovim config:

```lua
vim.api.nvim_create_autocmd({ "VimEnter", "VimResume" }, {
  group = vim.api.nvim_create_augroup("KittySetVarVimEnter", { clear = true }),
  callback = function()
    io.stdout:write("\x1b]1337;SetUserVar=in_editor=MQo\007")
  end,
})

vim.api.nvim_create_autocmd({ "VimLeave", "VimSuspend" }, {
  group = vim.api.nvim_create_augroup("KittyUnsetVarVimLeave", { clear = true }),
  callback = function()
    io.stdout:write("\x1b]1337;SetUserVar=in_editor\007")
  end,
})
```

Then I tell Kitty to not handle the keys I want to pass through when `in_editor` is set by doing this in `kitty.conf`:

```
map --when-focus-on var:in_editor cmd+c
map --when-focus-on var:in_editor cmd+v
map --when-focus-on var:in_editor cmd+x
```

Then I can set up Neovim key mappings like this:

```lua
vim.keymap.set("x", "<d-x>", '"*d', { desc = "Cut to system clipboard" })
vim.keymap.set("x", "<d-c>", '"*ygv', { desc = "Copy to system clipboard" })
vim.keymap.set({ "n", "x" }, "<d-v>", '"*p', { desc = "Paste from system clipboard" })
vim.keymap.set("i", "<d-v>", '<esc>"*pa', { desc = "Paste from system clipboard" })
vim.keymap.set("c", "<d-v>", "<c-R>*", { desc = "Paste from system clipboard" })
```

**Comment by rhodes-b** [comment]
Ghostty plans on being configurable by the command line but as is this is currently not possible. 
#2872 
above seems related, not sure if there are any other discussions on the topic (but I do know that this is planned for sure)
